include/deal.II/base/function_lib.h

```diff
@@ -930,25 +935,71 @@ namespace Functions
      *
      * If an argument <tt>select</tt> is given and not -1, the cut-off
      * function will be non-zero for this component only.
+     *
+     * If the argument @p integrate_to_one is set to true, then the value of
+     * the function is rescaled whenever a new radius is set.
      */
     CutOffFunctionBase(
       const double radius             = 1.,
       const Point<dim>                = Point<dim>(),
       const unsigned int n_components = 1,
-      const unsigned int select       = CutOffFunctionBase<dim>::no_component);
+      const unsigned int select       = CutOffFunctionBase<dim>::no_component,
+      const bool         integrate_to_one       = false,
+      const double       unitary_integral_value = 1.0);
+
+    /**
+     * Virtual destructor.
+     */
+    virtual ~CutOffFunctionBase() = default;
 
     /**
      * Move the center of the ball to new point <tt>p</tt>.
+     *
+     * @deprecated Use set_center() instead.
      */
-    void
+    DEAL_II_DEPRECATED void
     new_center(const Point<dim> &p);
 
     /**
      * Set the radius of the ball to <tt>r</tt>.
+     *
+     * @deprecated Use set_radius() instead.
      */
-    void
+    DEAL_II_DEPRECATED void
     new_radius(const double r);
 
+    /**
+     * Set the center of the ball to the point @p p.
+     */
+    virtual void
+    set_center(const Point<dim> &p);
+
+    /**
+     * Set the radius of the ball to @p r
+     *
+     * @deprecated Use set_radius() instead.
+     */
+    virtual void
+    set_radius(const double r);
```

Comment:
This is the new function. You want to remove the `@deprecated` line

Reply:
ah, well spotted. :)